Gem/Code/Source/HighInstanceExampleComponent.cpp

```diff
@@ -0,0 +1,309 @@
+        {
+            AZ::Render::MaterialAssignment& defaultAssignment = materials[AZ::Render::DefaultMaterialAssignmentId];
+            defaultAssignment = {};
+
+            if (instanceData.m_materialAssetId.IsValid())
+            {
+                defaultAssignment.m_materialAsset.Create(instanceData.m_materialAssetId);
+                defaultAssignment.m_materialInstance = AZ::RPI::Material::FindOrCreate(defaultAssignment.m_materialAsset);
+
+                // cache the material when its loaded
+                m_cachedMaterials.insert(defaultAssignment.m_materialAsset);
+            }
+
+            if (instanceData.m_modelAssetId.IsValid())
+            {
+                AZ::Data::Asset<AZ::RPI::ModelAsset> modelAsset;
+                modelAsset.Create(instanceData.m_modelAssetId);
+
+                instanceData.m_meshHandle = GetMeshFeatureProcessor()->AcquireMesh(AZ::Render::MeshHandleDescriptor{ modelAsset }, materials);
+                GetMeshFeatureProcessor()->SetTransform(instanceData.m_meshHandle, instanceData.m_transform);
+            }
+        }
+
+        ScriptRunnerRequestBus::Broadcast(&ScriptRunnerRequests::ResumeScript);
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void HighInstanceTestComponent::DestroyLatticeInstances()
+    {
+        DestroyHandles();
+        m_modelInstanceData.clear();
+    }
+
+    void HighInstanceTestComponent::DestroyHandles()
+    {
+        for (ModelInstanceData& instanceData : m_modelInstanceData)
+        {
+            GetMeshFeatureProcessor()->ReleaseMesh(instanceData.m_meshHandle);
+            instanceData.m_meshHandle = {};
+        }
+    }
+
+    AZ::Data::AssetId HighInstanceTestComponent::GetRandomModelId() const
+    {
+        auto& modelAllowlist = m_modelBrowser.GetPinnedAssets();
+
+        if (modelAllowlist.size())
+        {
+            const size_t randomModelIndex = rand() % modelAllowlist.size();
+            return modelAllowlist[randomModelIndex].m_assetId;
+        }
+        else
+        {
+            return AZ::RPI::AssetUtils::GetAssetIdForProductPath("testdata/objects/cube/cube.azmodel", AZ::RPI::AssetUtils::TraceLevel::Error);
+        }
+    }
+
+    AZ::Data::AssetId HighInstanceTestComponent::GetRandomMaterialId() const
+    {
+        auto& materialAllowlist = m_materialBrowser.GetPinnedAssets();
+
+        if (materialAllowlist.size())
+        {
+            const size_t randomMaterialIndex = rand() % materialAllowlist.size();
+            return materialAllowlist[randomMaterialIndex].m_assetId;
+        }
+        else
+        {
+            return AZ::RPI::AssetUtils::GetAssetIdForProductPath(DefaultPbrMaterialPath, AZ::RPI::AssetUtils::TraceLevel::Error);
+        }
+    }
+
+
+    void HighInstanceTestComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint scriptTime)
+    {
+        AZ_TRACE_METHOD();
+
+        if (m_updateTransformEnabled)
```

Comment:
I wonder (given the reason for this sample is profilling) if it makes sense to introduce a sleep here for the same amount of time as we think a modern open AAA game takes for gameplay. Basically we dont have a sample that stress tests gameplay/sim and it maybe a good idea to simulate that time here. It will then force us to think about moving Rendering to another thread sooner rather than later.

Reply:
Increasing the range of the sliders works for most of the Atom team, but the request here was to have something that any QA representative could trivially run and see how Atom's performance has changed. So the aim is to have a simple test that stresses Atoms cpu (not rendering) performance independent of anything else.

Reply:
This test is wholly about Atom's cpu perf. I will make a separate test in the AutomatedTesting project that pushes how Atom performs as part of O3DE